web/src/portico/team.ts

```diff
@@ -1,9 +1,31 @@
 import $ from "jquery";
 import _ from "lodash";
 
-import {page_params} from "../page_params";
+import * as blueslip from "../blueslip";
+
+const repository_name = [
```

Comment:
This is a bad name -- this is a list of repository names, and the variable name should express that. Probably `all_repository_names` would be a fine name. We should also not duplicate this list -- can't we compute it from `repo_name_to_tab_name`?

Reply:
Yeah, I agree that `repository_name` is a bad name for what it represents, I will fix that.
Coming on to not duplicating the list, it is possible to generate the type from the keys of the `repo_name_to_tab_name` object using the below type and then we can remove the redundant array of keys.
```
type RepositoryNameType = keyof typeof repo_name_to_tab_name;
```
The above type has all the keys of the object ([link to TS playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBATgUwA4gPpgIYFsGqmqDAI3WwRgF4YBvAKBgZgC8BXAGwEskAuGAIggI4ANyF8ANPUZ9WnJAFoAJgggBrfEj68+ytRolSGM9l3lYQRDmwRb+5y9YON+SAJ5QAFuHmzTGJBy2fG6e3r4K-oGSzsZy8gBWEEHhCUnRjOFEIFDamdlO0ilQQlgcmGxBxXCl5QVGKRwgEPLWAOYYwK5BdfwpGGCKcCAcit2SAL4A3LS0UK5I5ABKyE0c+HCuAHJkACrz5FSqCK4gAGYwcwtn8CukOHgExHcIhs4wAPTvAHoA-EA)) but since it's a type its not iterable. I tried finding way to convert this type to array but looks like there is no safe way out there to do that.
And Since we are iterating through the keys (at lines `46` and `119`) of the `repo_name_to_tab_name`  object only we don't get any type-checking there because then we have to use `Object.keys()` which only gives `string` data while we need them of the union type created before. We can use the `as` operator to fix this but that is not a very good option.
Hence, we can compute the list but then the iteration of keys won't be typesafe.